src/lsm/binary_search.zig

```diff
@@ -129,13 +129,148 @@ pub inline fn binary_search_keys(
     key: Key,
     comptime config: Config,
 ) BinarySearchResult {
-    const index = binary_search_keys_raw(Key, compare_keys, keys, key, config);
+    const index = binary_search_keys_upsert_index(Key, compare_keys, keys, key, config);
     return .{
         .index = index,
         .exact = index < keys.len and compare_keys(keys[index], key) == .eq,
     };
 }
 
+pub const BinarySearchRangeUpsertIndexes = struct {
+    start: u32,
+    end: u32,
+};
+
+pub inline fn binary_search_keys_range_upsert_indexes(
+    comptime Key: type,
+    comptime compare_keys: fn (Key, Key) callconv(.Inline) math.Order,
+    keys: []const Key,
+    key_min: Key,
+    key_max: Key,
+    comptime config: Config,
+) BinarySearchRangeUpsertIndexes {
+    return binary_search_values_range_upsert_indexes(
+        Key,
+        Key,
+        struct {
+            inline fn key_from_key(k: *const Key) Key {
+                return k.*;
+            }
+        }.key_from_key,
+        compare_keys,
+        keys,
+        key_min,
+        key_max,
+        config,
+    );
+}
+
+pub inline fn binary_search_values_range_upsert_indexes(
```

Comment:
Could you add a short doc comment here? In particular, looking just at the signature, the meaning of `key_min` / `key_max` is not immediately obvious to me.

Reply:
Ahhhh, this is not `equal_range`, we are looking for two keys at the same time. Still worth a short comment